File: service/src/migration/migrateDB.py
from pymongo import MongoClient
from bson.objectid import ObjectId
from dateutil.parser import parse

def migrate():
    uri = "CONNECTION_STRING/DBNAME?retryWrites=true&w=majority" # replace with your MongoDB connection string
    client = MongoClient(uri)

    try:
        print('Connected to MongoDB')

        userCol = client['DBNAME']['user'] # replace with your database and collection names
        print('Fetched user collection')

        cursor = userCol.find({})
        print('Created cursor for user documents')

        count = 0
        for doc in cursor:
            createTime = parse(doc['createTime'])
            verifyTime = parse(doc['verifyTime']) if 'verifyTime' in doc else None
            updateTime = parse(doc['updateTime']) if 'updateTime' in doc else None
            if createTime or verifyTime or updateTime:
                userCol.update_one({ '_id': ObjectId(doc['_id']) }, { '$set': { 'createTime': createTime, 'verifyTime': verifyTime, 'updateTime': updateTime } })
                count += 1
                print(f'Updated document {count}')

        print(f'Updated {count} documents')
    finally:
        client.close()
        print('Closed MongoDB connection')

# Call the migrate function
migrate()

File: service/src/migration/migrateDB.ts
import { MongoClient, ObjectId } from 'mongodb'

async function migrate() {
  const uri = 'CONNECTION_STRING/DBNAME?retryWrites=true&w=majority' // replace with your MongoDB connection string
  const client = new MongoClient(uri)

  try {
    await client.connect()
    console.log('Connected to MongoDB')

    const userCol = client.db('DBNAME').collection('user') // replace with your database and collection names
    console.log('Fetched user collection')

    const cursor = userCol.find({})
    console.log('Created cursor for user documents')

    let count = 0
    while (await cursor.hasNext()) {
      const doc = await cursor.next()
      const createTime = new Date(doc.createTime)
      const verifyTime = doc.verifyTime ? new Date(doc.verifyTime) : null
      const updateTime = doc.updateTime ? new Date(doc.updateTime) : null
      if (!Number.isNaN(createTime.getTime()) || (verifyTime && !Number.isNaN(verifyTime.getTime())) || (updateTime && !Number.isNaN(updateTime.getTime()))) {
        await userCol.updateOne({ _id: new ObjectId(doc._id) }, { $set: { createTime, verifyTime, updateTime } })
        count++
        console.log(`Updated document ${count}`)
      }
    }
    console.log(`Updated ${count} documents`)
  }
  finally {
    await client.close()
    console.log('Closed MongoDB connection')
  }
}

// Call the migrate function
migrate().catch(console.error)
